### PR TITLE
[Communication] - PhoneNumberClient - Adding samples for phone number administration

### PR DESCRIPTION
**Pull request by pavelprystinka**
Update from Azure

### CODE REVIEW

**Comment by DominikMe** [comment]
Cool stuff, can we split this PR into two? One for live tests, one for samples?

File: sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+            var geographicPhonePlanGroup = phonePlanGroups.First(group => group.PhoneNumberType == PhoneNumberType.Geographic);
+
+            var phonePlans = client.GetPhonePlans(countryCode, geographicPhonePlanGroup.PhonePlanGroupId, locale);
+            var phonePlanId = phonePlans.First().PhonePlanId;
+
+            #region Snippet:GetPhonePlanLocationOptions
+
+            var locationOptionsResponse = client.GetPhonePlanLocationOptions(countryCode, geographicPhonePlanGroup.PhonePlanGroupId, phonePlanId);
+
+            void printLocationOption(LocationOptions locationOptions)
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs:286]: nit: Capitalization of the method is not correct.

File: sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+            }
+            #endregion Snippet:GetAllSupportedCountriesAsync
+        }
+
+        [Test]
+        [SyncOnly]
+        public void GetPhonePlanLocationOptions()
+        {
+            var client = CreateClient(false);
+            var locale = "en-US";
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs:273]: `const` instead of `var`. Same for the country code and other methods

File: sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+            var phonePlans = client.GetPhonePlans(countryCode, geographicPhonePlanGroup.PhonePlanGroupId, locale);
+            var geographicPhonePlan = phonePlans.First();
+            var geographicPhonePlanGroupId = geographicPhonePlanGroup.PhonePlanGroupId;
+            var geographicPhonePlanId = geographicPhonePlan.PhonePlanId;
+
+            #region Snippet:GeographicalAreaCodes
+            var locationOptionsResponse = client.GetPhonePlanLocationOptions(countryCode, geographicPhonePlanGroupId, geographicPhonePlanId);
+            var state = locationOptionsResponse.Value.LocationOptions.Options.First();
+
+            var locationOptionsQueries = new List<LocationOptionsQuery>{
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs:359]: nit: `{` on the new line. I see similar issue in other places too.

File: sdk/communication/Azure.Communication.Administration/tests/PhoneNumberAdministrationClient/PhoneNumberAdministrationClientLiveTests.cs [hunk 4]
```diff
@@ -153,6 +153,10 @@ public async Task GetAreaCodesForPlan(string? locale)
         [AsyncOnly]
         public async Task CreateReservationErrorState(string? locale)
         {
+            if (!IncludePhoneNumberLiveTests)
+                Assert.Ignore("Include phone number live tests flag is off.");
+
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Administration/tests/PhoneNumberAdministrationClient/PhoneNumberAdministrationClientLiveTests.cs:158]: nit: extra new line

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+
+while (true)
+{
+    newPhoneNumberReservationOperation.UpdateStatus();
+    if (newPhoneNumberReservationOperation.HasCompleted)
+    {
+        break;
+    }
+
+    Thread.Sleep(1000);
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:161]: @christothes is there a better recommended way to have sync tests for long running operations?

File: sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs [hunk 6]
```diff
@@ -0,0 +1,471 @@
...
+                Console.WriteLine($"Phone number: {phoneNumber.PhoneNumber}, activation state: {phoneNumber.ActivationState}");
+            }
+            #endregion Snippet:ListAcquiredPhoneNumbersAsync
+
+            var acquiredPhoneNumber = reservationOperation.Value.PhoneNumbers.First();
+
+            #region Snippet:ReleasePhoneNumbersAsync
+            //@@var acquiredPhoneNumber = "<acquired_phone_number>";
+            var phoneNumbers = new List<PhoneNumber> { new PhoneNumber(acquiredPhoneNumber) };
+            var phoneNumberReleaseOperation = client.StartReleasePhoneNumbers(phoneNumbers);
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs:102]: you can inline:
```C#
client.StartReleasePhoneNumbers(new [] { new PhoneNumber(acquiredPhoneNumber) });
```

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md (hunk 5 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:161]: @christothes @RezaJooyandeh just a reminder that this test code is copied over MD sample file

File: sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs (hunk 6 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs:102]: @RezaJooyandeh I would prefer to leave second line simpler as this is for the MD sample and it is somewhat easier to those who is glancing over it to catch that release starts by passing some phoneNumbers, whatever type those are.
Thoughts?

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md (hunk 5 above)
Comment (DominikMe) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:162]: Let's follow how KeyVault has written the code. https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/keyvault/Azure.Security.KeyVault.Keys/samples/Sample1_HelloWorld.md#purging-a-deleted-key

```csharp
while (!newPhoneNumberReservationOperation.HasCompleted)
{
    Thread.Sleep(2000);

    operation.UpdateStatus();
}
```

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+## Persist reserve phone numbers operation
+
+You can persist phone number reservation operation Id so that you can get back later to check operation status.
+
+```C# Snippet:PersistReservePhoneNumbersOperation
+var reservationId = reservationOperation.Id;
+
+// persist reservationOperationId and then continue with new operation
+
+var newPhoneNumberReservationOperation = new PhoneNumberReservationOperation(client, reservationId);
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:152]: Can we shorten the name? Maybe just `operation` or `reserveOperation`.

The context makes it clear what the operation is about and it can't be confused with another operation, that's why I think a short and less precise name reads better.

**Review by DominikMe — CHANGES_REQUESTED** [review]
Great job on the samples and writing the README! 🐱‍💻

Added comments with fixes and suggestions on the text. Also weighed in on the infinite loop for synchronously polling a LRO.

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClientAsync.md [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+
+## Get area codes
+
+Fetching area codes for geographic phone plans will require the the location options queries set. You must include the chain of geographic locations traversing down the location options object returned by the GetPhonePlanLocationOptions.
+
+```C# Snippet:GeographicalAreaCodesAsync
+var locationOptionsResponse = await client.GetPhonePlanLocationOptionsAsync(countryCode, geographicPhonePlanGroupId, geographicPhonePlanId);
+var state = locationOptionsResponse.Value.LocationOptions.Options.First();
+
+var locationOptionsQueries = new List<LocationOptionsQuery>{
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClientAsync.md:85]: `{` on the new line

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md (hunk 7 above)
Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:152]: +1 on `reserveOperation`

Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:152]: +1 on `reserveOperation`

File: sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs (hunk 2 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs]: Done

File: sdk/communication/Azure.Communication.Administration/tests/PhoneNumberAdministrationClient/PhoneNumberAdministrationClientLiveTests.cs (hunk 4 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/tests/PhoneNumberAdministrationClient/PhoneNumberAdministrationClientLiveTests.cs:158]: Done

File: sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs (hunk 6 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs]: I have added a new method that takes a single phone number and used it on a sample

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md [hunk 9]
```diff
@@ -0,0 +1,215 @@
+# Searching, reserving, purchasing, releasing phone numbers
+
+This sample demonstrates how to search, reserve, purchase and release phone numbers in Azure Communication Services.
+To get started, you'll need a URI to an Azure Communication Services. See the [README](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/communication/Azure.Communication.Administration/README.md) for links and instructions.
+
+## Creating a PhoneNumberAdministrationClient
+
+To create a new `PhoneNumberAdministrationClient` you need a connection string to the Azure Communication Services resource that you can get from the Azure Portal once you create a relevant resource.
```
[the remark this answers is not in the document]
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:8]: Done

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+
+```C# Snippet:CreatePhoneNumberAdministrationClient
+// Get a connection string to our Azure Communication resource.
+var connectionString = "<connection_string>";
+var client = new PhoneNumberAdministrationClient(connectionString);
+```
+
+## Listing all supported countries
+
+In order to acquire a phone number you will need to know if Azure Communication Services are available in particular country. You can do that by retrieving a list of supported countries.
```
[the remark this answers is not in the document]
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:20]: Done

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+        foreach (var subLocationOption in locationOption.LocationOptions)
+            PrintLocationOption(subLocationOption);
+    }
+}
+PrintLocationOption(locationOptionsResponse.Value.LocationOptions);
+```
+
+## Get area codes
+
+Fetching area codes for geographic phone plans will require the the location options queries set. You must include the chain of geographic locations traversing down the location options object returned by the GetPhonePlanLocationOptions.
```
[the remark this answers is not in the document]
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:79]: Done

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md [hunk 12]
```diff
@@ -0,0 +1,215 @@
...
+
+var areaCodes = client.GetAllAreaCodes(geographicPhonePlan.LocationType.ToString(), countryCode, geographicPhonePlan.PhonePlanId, locationOptionsQueries);
+
+foreach (var areaCode in areaCodes.Value.PrimaryAreaCodes)
+{
+    Console.WriteLine($"Primary area code: {areaCode}");
+}
+```
+
+Area codes for toll free phone plans can be found in the plan.
```
[the remark this answers is not in the document]
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:107]: Done

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md [hunk 13]
```diff
@@ -0,0 +1,215 @@
...
+        break;
+    }
+
+    Thread.Sleep(1000);
+}
+```
+
+## Purchase phone numbers
+
+Phone numbers can be acquired by purchasing reservation.
```
[the remark this answers is not in the document]
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:168]: Done

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md [hunk 14]
```diff
@@ -0,0 +1,215 @@
...
+        break;
+    }
+
+    Thread.Sleep(1000);
+}
+```
+
+## Persist reserve phone numbers operation
+
+You can persist phone number reservation operation Id so that you can get back later to check operation status.
```
[the remark this answers is not in the document]
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:145]: Done

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md [hunk 15]
```diff
@@ -0,0 +1,215 @@
...
+
+foreach (var areaCode in tollFreePhonePlan.AreaCodes)
+{
+    Console.WriteLine($"Area code: {areaCode}");
+}
+```
+
+## Reserve phone numbers
+
+Phone numbers need to be reserved for purchase. Reservation is a long running operation that can be started by CreateReservationOptions function that returns an PhoneNumberReservationOperation object. PhoneNumberReservationOperation can be used to update status of the operation and to check for completeness.
```
[the remark this answers is not in the document]
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:121]: Done

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md [hunk 16]
```diff
@@ -0,0 +1,215 @@
...
+    {
+        break;
+    }
+
+    Thread.Sleep(1000);
+}
+```
+
+## Listing acquired phone numbers
+
```
[the remark this answers is not in the document]
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:186]: Done

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md [hunk 17]
```diff
@@ -0,0 +1,215 @@
...
+foreach (var areaCode in areaCodes.Value.PrimaryAreaCodes)
+{
+    Console.WriteLine($"Primary area code: {areaCode}");
+}
+```
+
+Area codes for toll free phone plans can be found in the plan.
+
+```C# Snippet:TollFreePlanAreCodes
```
[the remark this answers is not in the document]
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:109]: Done

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClientAsync.md [hunk 18]
```diff
@@ -0,0 +1,174 @@
...
+
+foreach (var areaCode in areaCodes.Value.PrimaryAreaCodes)
+{
+    Console.WriteLine($"Primary area code: {areaCode}");
+}
+```
+
+Area codes for toll free phone plans can be found in the plan.
+
+```C# Snippet:TollFreePlanAreCodesAsync
```
[the remark this answers is not in the document]
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClientAsync.md:108]: Done

File: sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs [hunk 19]
```diff
@@ -0,0 +1,472 @@
...
+            foreach (var areaCode in areaCodes.Value.PrimaryAreaCodes)
+            {
+                Console.WriteLine($"Primary area code: {areaCode}");
+            }
+            #endregion Snippet:GeographicalAreaCodesAsync
+        }
+
+        [Test]
+        [SyncOnly]
+        public void TollFreePlanAreCodes()
```
[the remark this answers is not in the document]
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs:435]: Done

File: sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs [hunk 20]
```diff
@@ -0,0 +1,472 @@
...
+        {
+            var client = CreateClient(false);
+            const string locale = "en-US";
+            const string countryCode = "US";
+
+            var phonePlanGroups = client.GetPhonePlanGroups(countryCode, locale);
+            var tollFreePhonePlanGroup = phonePlanGroups.First(group => group.PhoneNumberType == PhoneNumberType.TollFree);
+            var tollFreePhonePlanGroupId = tollFreePhonePlanGroup.PhonePlanGroupId;
+            #region Snippet:TollFreePlanAreCodes
```
[the remark this answers is not in the document]
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs:444]: Done

File: sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs [hunk 21]
```diff
@@ -0,0 +1,472 @@
...
+            {
+                Console.WriteLine($"Area code: {areaCode}");
+            }
+
+            #endregion Snippet:TollFreePlanAreCodes
+        }
+
+        [Test]
+        [AsyncOnly]
+        public async Task TollFreePlanAreCodesAsync()
```
[the remark this answers is not in the document]
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs:459]: Done

File: sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs [hunk 22]
```diff
@@ -0,0 +1,472 @@
...
+
+            var phonePlans = client.GetPhonePlans(countryCode, tollFreePhonePlanGroupId, locale);
+            var tollFreePhonePlan = phonePlans.First();
+
+            foreach (var areaCode in tollFreePhonePlan.AreaCodes)
+            {
+                Console.WriteLine($"Area code: {areaCode}");
+            }
+
+            #endregion Snippet:TollFreePlanAreCodes
```
[the remark this answers is not in the document]
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs:454]: Done

File: sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs [hunk 23]
```diff
@@ -0,0 +1,472 @@
...
+        public async Task TollFreePlanAreCodesAsync()
+        {
+            var client = CreateClient(false);
+            const string locale = "en-US";
+            const string countryCode = "US";
+
+            var phonePlanGroups = client.GetPhonePlanGroups(countryCode, locale);
+            var tollFreePhonePlanGroup = phonePlanGroups.First(group => group.PhoneNumberType == PhoneNumberType.TollFree);
+
+            #region Snippet:TollFreePlanAreCodesAsync
```
[the remark this answers is not in the document]
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs:468]: Done

File: sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs [hunk 24]
```diff
@@ -0,0 +1,472 @@
...
+
+            var phonePlans = client.GetPhonePlansAsync(countryCode, tollFreePhonePlanGroup.PhonePlanGroupId, locale);
+            var tollFreePhonePlan = (await phonePlans.ToEnumerableAsync()).First();
+
+            foreach (var areaCode in tollFreePhonePlan.AreaCodes)
+            {
+                Console.WriteLine($"Area code: {areaCode}");
+            }
+
+            #endregion Snippet:TollFreePlanAreCodesAsync
```
[the remark this answers is not in the document]
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs:478]: Done

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClientAsync.md (hunk 8 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClientAsync.md:85]: Done

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md (hunk 7 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:152]: I made it newReserveOperation to stress that is a new object created with reservationId
Also there is reserveOperation variable in the same test where code snippet is coming

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md (hunk 5 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md]: Done

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md (hunk 7 above)
Comment (DominikMe) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:148]: Usually one would only prefix variable names with `new` if they are a new replacement for something existing, for example an event handler with `newValue`. Otherwise every variable that holds a constructed object would start with `new`. In general, we're preferring short names over encoding types in the name, which is a practice that comes from Hungarian notation among others and is now discouraged.

To avoid a name clash with the other `reserveOperation` you can extract a function to have a new context.

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md [hunk 25]
```diff
@@ -0,0 +1,200 @@
...
+        foreach (var subLocationOption in locationOption.LocationOptions)
+            PrintLocationOption(subLocationOption);
+    }
+}
+PrintLocationOption(locationOptionsResponse.Value.LocationOptions);
+```
+
+## Get area codes
+
+Fetching area codes for geographic phone plans will require the location options queries set. You must include the chain of geographic locations traversing down the location options object returned by the GetPhonePlanLocationOptions.
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:79]:
```suggestion
Fetching area codes for geographic phone plans will require the location options queries set. You must include the chain of geographic locations traversing down the location options object returned by the `GetPhonePlanLocationOptions`.
```

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md [hunk 26]
```diff
@@ -0,0 +1,200 @@
...
+    newPhoneNumberReservationOperation.UpdateStatus();
+}
+```
+
+## Purchase phone numbers
+
+Phone numbers can be acquired through purchasing a reservation.
+
+```C# Snippet:StartPurchaseReservation
+var reservationPurchaseOperation = client.StartPurchaseReservation(reservationId);
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:163]: We can shorten this to `purchaseOperation`

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md [hunk 27]
```diff
@@ -0,0 +1,200 @@
...
+}
+```
+
+## Release phone numbers
+
+If you no longer need a phone number you can release it.
+
+```C# Snippet:ReleasePhoneNumbers
+var acquiredPhoneNumber = "<acquired_phone_number>";
+var phoneNumberReleaseOperation = client.StartReleasePhoneNumber(new PhoneNumber(acquiredPhoneNumber));
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:192]: Let's shorten to `releaseOperation`

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClientAsync.md [hunk 28]
```diff
@@ -0,0 +1,176 @@
...
+var newReserveOperation = new PhoneNumberReservationOperation(client, reservationId);
+await newReserveOperation.WaitForCompletionAsync();
+```
+
+## Purchase phone numbers
+
+Phone numbers can be acquired through purchasing a reservation.
+
+```C# Snippet:StartPurchaseReservationAsync
+var reservationPurchaseOperation = await client.StartPurchaseReservationAsync(reservationId);
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClientAsync.md:151]: Let's shorten the name to `purchaseOperation`.

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClientAsync.md [hunk 29]
```diff
@@ -0,0 +1,176 @@
...
+}
+```
+
+## Release phone numbers
+
+If you no longer need a phone number you can release it.
+
+```C# Snippet:ReleasePhoneNumbersAsync
+var acquiredPhoneNumber = "<acquired_phone_number>";
+var phoneNumberReleaseOperation = client.StartReleasePhoneNumber(new PhoneNumber(acquiredPhoneNumber));
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClientAsync.md:174]: Let's shorten the name to `releaseOperation`.

File: sdk/communication/Azure.Communication.Administration/src/PhoneNumberAdministrationClient.cs [hunk 30]
```diff
@@ -554,6 +554,25 @@ public virtual Response<PhoneNumberRelease> GetReleaseById(string releaseId, Can
             }
         }
 
+        /// <summary> Starts a release for the given phone numbers. </summary>
+        /// <param name="phoneNumber"> The phone number in the release request. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <returns> A <see cref="ReleasePhoneNumberOperation"/>. </returns>
+        public virtual async Task<ReleasePhoneNumberOperation> StartReleasePhoneNumberAsync(PhoneNumber phoneNumber, CancellationToken cancellationToken = default)
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Administration/src/PhoneNumberAdministrationClient.cs:561]: I think it's fine adding another overload for a single number. What do you think @RezaJooyandeh?

Ideally we would add the overload for each language to stay consistent. Also, a change like this should happen in its own PR (see my other comment about strict single-purpose PRs).

**Review by DominikMe — CHANGES_REQUESTED** [review]
Some outstanding naming nits and question to Reza about adding the overload.

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md (hunk 7 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:148]: Renamed it to reserveOperation

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md (hunk 25 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:79]: Done

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClientAsync.md (hunk 29 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClientAsync.md:174]: Done

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClientAsync.md (hunk 28 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClientAsync.md:151]: Done

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md (hunk 27 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:192]: Done

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md (hunk 26 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:163]: Done

File: sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md (hunk 25 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md:79]: Done

**Review by DominikMe — APPROVED** [review]
Synced up offline, ready to merge after addressing the comments with the next commit.
